src/map/entities/charentity.cpp

```diff
@@ -2097,6 +2097,15 @@ void CCharEntity::OnRangedAttack(CRangeState& state, action_t& action)
         hitCount = 3;
     }
 
+    if (state.IsOutOfRange())
+    {
+        hitCount                = 1;
```

Comment:
This should probably be 0

`1 <= 1` is true so it would do one loop

`for (uint8 i = 1; i <= hitCount; ++i)`